api/application.go

```diff
@@ -142,7 +142,7 @@ func (h ApplicationHandler) Delete(ctx *gin.Context) {
 		h.deleteFailed(ctx, result.Error)
 		return
 	}
-	result = h.DB.Select(clause.Associations).Delete(m)
```

Comment:
Rely on cascade delete.
This can be dangerous because it will delete everything associated with the application with is not always correct.